plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemoryUsage.hs

```diff
@@ -226,6 +227,53 @@ instance ExMemoryUsage () where
     memoryUsage () = singletonRose 1
     {-# INLINE memoryUsage #-}
 
+{- | Calculate a 'CostingInteger' for the size of the given 'Integer', measured in
+64-bit words.  This is the default size measure for `Integer`s.
+-}
+memoryUsageInteger :: Integer -> CostingInteger
+-- integerLog2# is unspecified for 0 (but in practice returns -1)
+-- ^ This changed with GHC 9.2: it now returns 0.  It's probably safest if we
+-- keep this special case for the time being though.
+memoryUsageInteger 0 = 1
+-- Assume 64 Int
+memoryUsageInteger i = fromIntegral $ I# (integerLog2# (abs i) `quotInt#` integerToInt 64) + 1
+-- So that the produced GHC Core doesn't explode in size, we don't win anything by inlining this
+-- function anyway.
+{-# OPAQUE memoryUsageInteger #-}
+
+instance ExMemoryUsage Integer where
+    memoryUsage i = singletonRose $ memoryUsageInteger i
+    {-# INLINE memoryUsage #-}
+
+instance ExMemoryUsage Natural where
+    -- Same as Integer since we are going via Integer
+    memoryUsage n = memoryUsage $ toInteger n
+    {-# INLINE memoryUsage #-}
+
+instance ExMemoryUsage Word8 where
+    memoryUsage _ = singletonRose 1
+    {-# INLINE memoryUsage #-}
+
+{- | Calculate a 'CostingInteger' for the size of the given 'Integer', measured in
+   8-bit bytes. -}
+memoryUsageBytes :: Integer -> CostingInteger
```

Comment:
I added this because using the standard measure would give very coarse-grained results and this is much more accurate.  Now I'm worried that it's jsut adding more complexity.  Maybe we should go for inaccurate but simple.